drivers/modem/gsm_ppp.c

```diff
@@ -174,10 +174,11 @@ static void gsm_rx(struct gsm_modem *gsm)
 	LOG_DBG("starting");
 
 	while (true) {
-		(void)k_sem_take(&gsm->gsm_data.rx_sem, K_FOREVER);
+		/* Wait for incoming data */
+		modem_iface_uart_rx_wait(&gsm->context.iface, K_FOREVER);
 
-		/* The handler will listen AT channel */
-		gsm->context.cmd_handler.process(&gsm->context.cmd_handler,
+		/* Process data */
+		modem_cmd_handler_process(&gsm->context.cmd_handler,
```

Comment:
since the args arent aligned anymore, probably can just make it into a single line